### PR TITLE
COMMERCE-10560

### PR DESCRIPTION
**Pull request by Brunno0liveira**

### CODE REVIEW

**Comment by Brunno0liveira** [comment]
ci:test:commerce

**Comment by Brunno0liveira** [comment]
ci:test:sf

**Comment by Brunno0liveira** [comment]
ci:test:commerce

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase [hunk 1]
```diff
@@ -17,6 +17,215 @@ definition {
...
+				newAddressCountry = "United States",
+				newAddressName = "Address Name",
+				newAddressZipCode = "Test Zip Code",
+				productName = "U-Joint",
+				productQuantity = "2",
+				productUnitPrice = "24",
+				shippingMethod = "Standard");
+		}
+
+		task ("Then product remains available to be added to the cart and buyer user can place a second order") {
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase:207]:
```suggestion
		task ("Then the product can still be added to the cart and buyer user can place a second order") {
```

**Review by lorenzolapescara — CHANGES_REQUESTED** [review]
@Brunno0liveira test descriptions and names are missing the details about the Allow Back Order configuration. Can you add the details to the description and try to change the test names to be compliant with the back order configuration? Thanks

**Comment by Brunno0liveira** [comment]
ci:test:sf

**Comment by Brunno0liveira** [comment]
ci:test:sf

**Comment by Brunno0liveira** [comment]
ci:test:commerce

**Comment by Brunno0liveira** [comment]
ci:test:commerce

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase [hunk 2]
```diff
@@ -17,9 +17,9 @@ definition {
 		CommerceConfiguration.commerceTearDown();
 	}
 
-	@description = "This is a test for COMMERCE-10542. Maximum quantity can be added to cart when order is higher than stock quantity"
+	@description = "This is a test for COMMERCE-10560. Allowed quantities can be added to cart when order is higher than stock quantity when back orders are enable"
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase:20]:
```suggestion
	@description = "This is a test for COMMERCE-10560. Allowed quantities can be added to the cart when order quantity is higher than the stock quantity and back order is enabled"
```

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase [hunk 3]
```diff
@@ -17,9 +17,9 @@ definition {
...
 	@priority = "3"
-	test CanAddToCartMaximumProductQuantityWhenOrderItemQuantityIsHigherThanStockQuantity {
+	test CanAddToCartAllowedQuantitiesWhenOrderItemQuantityIsHigherThanStockQuantityAndBackOrdersAreEnable {
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase:22]:
```suggestion
	test CanAddToCartAllowedQuantitiesWhenOrderItemQuantityIsHigherThanStockQuantityAndBackOrderIsEnabled {
```

**Review by lorenzolapescara — CHANGES_REQUESTED** [review]
@Brunno0liveira please check this comments and look for similar errors in the test descriptions and names. Thanks

**Comment by Brunno0liveira** [comment]
ci:test:sf

**Comment by Brunno0liveira** [comment]
ci:test:commerce

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase [hunk 4]
```diff
@@ -89,9 +302,9 @@ definition {
 		}
 	}
 
-	@description = "This is a test for COMMERCE-10542. Maximum quantity can be added to cart when order is lower than stock quantity"
+	@description = "This is a test for COMMERCE-10542. Maximum quantity can be added to the cart when order quantity is lower than stock quantity and back orders is enabled"
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase:305]:
```suggestion
	@description = "This is a test for COMMERCE-10542. Maximum quantity can be added to the cart when order quantity is lower than stock quantity and back order is enabled"
```

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase [hunk 5]
```diff
@@ -89,9 +302,9 @@ definition {
...
 	@priority = "3"
-	test CanAddToCartMaximumProductQuantityWhenOrderItemQuantityIsLowerThanStockQuantity {
+	test CanAddToCartMaximumProductQuantityWhenOrderItemQuantityIsLowerThanStockQuantityAndBackOrdersIsEnable {
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase:307]:
```suggestion
	test CanAddToCartMaximumQuantityWhenOrderItemQuantityIsLowerThanStockQuantityAndBackOrderIsEnabled {
```

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase [hunk 6]
```diff
@@ -182,9 +393,9 @@ definition {
 		}
 	}
 
-	@description = "This is a test for COMMERCE-10503. Minimum quantity can be added to cart when order is higher then stock quantity"
+	@description = "This is a test for COMMERCE-10503. Minimum quantity can be added to the cart when order quantity is higher then the stock quantity and back orders is enabled"
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase:396]:
```suggestion
	@description = "This is a test for COMMERCE-10503. Minimum quantity can be added to the cart when order quantity is higher than the stock quantity and back order is enabled"
```

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase [hunk 7]
```diff
@@ -182,9 +393,9 @@ definition {
...
 	@priority = "3"
-	test CanAddToCartMinimumProductQuantityWhenOrderItemQuantityIsHigherThanStockQuantity {
+	test CanAddToCartMinimumProductQuantityWhenOrderItemQuantityIsHigherThanStockQuantityAndBackOrdersIsEnable {
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase:398]:
```suggestion
	test CanAddToCartMinimumQuantityWhenOrderItemQuantityIsHigherThanStockQuantityAndBackOrderIsEnabled {
```

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase [hunk 8]
```diff
@@ -261,9 +470,9 @@ definition {
 		}
 	}
 
-	@description = "This is a test for COMMERCE-10503. Minimum quantity can be added to cart when order item quantity is lower than stock quantity"
+	@description = "This is a test for COMMERCE-10503. Minimum quantity can be added to the cart when order quantity is lower than the stock quantity and back orders is enabled"
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase:473]:
```suggestion
	@description = "This is a test for COMMERCE-10503. Minimum quantity can be added to the cart when order quantity is lower than the stock quantity and back order is enabled"
```

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase [hunk 9]
```diff
@@ -261,9 +470,9 @@ definition {
...
 	@priority = "3"
-	test CanAddToCartMinimumProductQuantityWhenOrderItemQuantityIsLowerThanStockQuantity {
+	test CanAddToCartMinimumProductQuantityWhenOrderItemQuantityIsLowerThanStockQuantityAndBackOrdersIsEnable {
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase:475]:
```suggestion
	test CanAddToCartMinimumQuantityWhenOrderItemQuantityIsLowerThanStockQuantityAndBackOrderIsEnabled {
```

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase [hunk 10]
```diff
@@ -361,9 +568,9 @@ definition {
 		}
 	}
 
-	@description = "This is a test for COMMERCE-10559. Multiple quantity can be added to the cart when order is higher than stock quantity"
+	@description = "This is a test for COMMERCE-10559. Multiple quantity can be added to the cart when order quantity is higher than the stock quantity and back orders is enabled"
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase:571]:
```suggestion
	@description = "This is a test for COMMERCE-10559. Multiple quantities can be added to the cart when order quantity is higher than the stock quantity and back order is enabled"
```

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase [hunk 11]
```diff
@@ -361,9 +568,9 @@ definition {
...
 	@priority = "3"
-	test CanAddToCartMultipleProductQuantityWhenOrderItemQuantityIsHigherThanStockQuantity {
+	test CanAddToCartMultipleProductQuantityWhenOrderItemQuantityIsHigherThanStockQuantityAndBackOrdersIsEnable {
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase:573]:
```suggestion
	test CanAddToCartMultipleQuantityWhenOrderItemQuantityIsHigherThanStockQuantityAndBackOrderIsEnabled {
```

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase [hunk 12]
```diff
@@ -433,9 +638,9 @@ definition {
 		}
 	}
 
-	@description = "This is a test for COMMERCE-10559. Multiple quantity can be added to cart when order is lower than stock quantity"
+	@description = "This is a test for COMMERCE-10559. Multiple quantity can be added to the cart when order quantity is lower than the stock quantity and back orders is enabled"
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase:641]:
```suggestion
	@description = "This is a test for COMMERCE-10559. Multiple quantities can be added to the cart when order quantity is lower than the stock quantity and back order is enabled"
```

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase [hunk 13]
```diff
@@ -433,9 +638,9 @@ definition {
...
 	@priority = "3"
-	test CanAddToCartMultipleProductQuantityWhenOrderItemQuantityIsLowerThanStockQuantity {
+	test CanAddToCartMultipleProductQuantityWhenOrderItemQuantityIsLowerThanStockQuantityAndBackOrdersIsEnable {
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase:643]:
```suggestion
	test  #CanAddToCartMultipleQuantityWhenOrderItemQuantityIsLowerThanStockQuantityAndBackOrderIsEnabled {
```

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase [hunk 14]
```diff
@@ -546,15 +749,193 @@ definition {
...
+				key_productName = "U-Joint",
+				locator1 = "CommerceAccelerators#ADD_TO_CART_BUTTON");
+		}
+
+		task ("Then it's not possible to add the order item to the cart") {
+			Alert.viewErrorMessage(errorMessage = "Danger:The specified quantity is unavailable.");
+		}
+	}
+
+	@description = "This is a test for COMMERCE-10560. Allowed quantities can be added to the cart until there's stock quantity and back orders is unabled"
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase:798]:
```suggestion
	@description = "This is a test for COMMERCE-10560. Allowed quantities can be added to the cart until there's stock quantity and back order is disabled"
```

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase [hunk 15]
```diff
@@ -546,15 +749,193 @@ definition {
...
+	@priority = "3"
+	test CannotAddToCartAllowedQuantitiesWhenStockQuantityIsLessAndBackOrdersIsUnable {
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase:800]:
```suggestion
	test CannotAddToCartAllowedQuantitiesWhenStockQuantityIsLessAndBackOrdersIsUnable {
```
```suggestion
	test CannotAddToCartAllowedQuantitiesWhenStockQuantityIsLessAndBackOrderIsDisabled {
```

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase [hunk 16]
```diff
@@ -546,15 +749,193 @@ definition {
...
+				key_productName = "U-Joint",
+				locator1 = "CommerceAccelerators#ADD_TO_CART_BUTTON");
+		}
+
+		task ("Then the order item will not be added to the cart") {
+			Alert.viewErrorMessage(errorMessage = "Danger:The specified quantity is unavailable.");
+		}
+	}
+
+	@description = "This is a test for COMMERCE-10542. Maximum quantity cannot be added to the cart when order quantity is higher than the stock quantity and back orders is unabled"
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase:912]:
```suggestion
	@description = "This is a test for COMMERCE-10542. Maximum quantity cannot be added to the cart when order quantity is higher than the stock quantity and back order is disabled"
```

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase [hunk 17]
```diff
@@ -546,15 +749,193 @@ definition {
...
+	@priority = "3"
+	test CannotAddToCartMaximumProductQuantityWhenOrderItemQuantityIsHigherThanStockQuantityAndBackOrdersIsUnable {
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase:914]:
```suggestion
	test CannotAddToCartMaximumQuantityWhenOrderItemQuantityIsHigherThanStockQuantityAndBackOrderIsDisabled {
```

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase [hunk 18]
```diff
@@ -592,9 +973,9 @@ definition {
 		}
 	}
 
-	@description = "This is a test for COMMERCE-10542. Maximum quantity can be added to cart until there's stock quantity "
+	@description = "This is a test for COMMERCE-10542. Maximum quantity can be added to the cart until there's stock quantity and back orders is unabled"
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase:976]:
```suggestion
	@description = "This is a test for COMMERCE-10542. Maximum quantity can be added to the cart until there's stock quantity and back order is disabled"
```

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase [hunk 19]
```diff
@@ -592,9 +973,9 @@ definition {
...
 	@priority = "3"
-	test CannotAddToCartMaximumProductQuantityWhenStockQuantityIsLess {
+	test CannotAddToCartMaximumProductQuantityWhenStockQuantityIsLessAndBackOrdersIsUnable {
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase:978]:
```suggestion
	test CannotAddToCartMaximumQuantityWhenStockQuantityIsLessAndBackOrderIsDisabled {
```

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase [hunk 20]
```diff
@@ -681,9 +1060,9 @@ definition {
 		}
 	}
 
-	@description = "This is a test for COMMERCE-10503. Minimum quantity cannot be added to cart when order is higher then stock quantity"
+	@description = "This is a test for COMMERCE-10503. Minimum quantity cannot be added to the cart when order quantity is higher then the stock quantity and back orders is unabled"
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase:1063]:
```suggestion
	@description = "This is a test for COMMERCE-10503. Minimum quantity cannot be added to the cart when order quantity is higher then the stock quantity and back order is disabled"
```

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase [hunk 21]
```diff
@@ -681,9 +1060,9 @@ definition {
...
 	@priority = "3"
-	test CannotAddToCartMinimumProductQuantityWhenOrderItemQuantityIsHigherThanStockQuantity {
+	test CannotAddToCartMinimumProductQuantityWhenOrderItemQuantityIsHigherThanStockQuantityAndBackOrdersIsUnable {
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase:1065]:
```suggestion
	test CannotAddToCartMinimumQuantityWhenOrderItemQuantityIsHigherThanStockQuantityAndBackOrderIsDisabled {
```

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase [hunk 22]
```diff
@@ -749,9 +1126,9 @@ definition {
 		}
 	}
 
-	@description = "This is a test for COMMERCE-10503. Minimum quantity can be added to cart until there's stock quantity "
+	@description = "This is a test for COMMERCE-10503. Minimum quantity can be added to the cart until there's stock quantity and back orders is unabled"
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase:1129]:
```suggestion
	@description = "This is a test for COMMERCE-10503. Minimum quantity can be added to the cart until there's stock quantity and back order is disabled"
```

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase [hunk 23]
```diff
@@ -749,9 +1126,9 @@ definition {
...
 	@priority = "3"
-	test CannotAddToCartMinimumProductQuantityWhenStockQuantityIsLess {
+	test CannotAddToCartMinimumProductQuantityWhenStockQuantityIsLessAndBackOrdersIsUnable {
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase:1131]:
```suggestion
	test CannotAddToCartMinimumQuantityWhenStockQuantityIsLessAndBackOrderIsDisabled {
```

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase [hunk 24]
```diff
@@ -838,9 +1213,9 @@ definition {
 		}
 	}
 
-	@description = "This is a test for COMMERCE-10559. Multiple quantity cannot be added to cart when order is higher than stock quantity"
+	@description = "This is a test for COMMERCE-10559. Multiple quantity cannot be added to the cart when order quantity is higher than the stock quantity and back orders is unable"
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase:1216]:
```suggestion
	@description = "This is a test for COMMERCE-10559. Multiple quantities cannot be added to the cart when order quantity is higher than the stock quantity and back order is disabled"
```

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase [hunk 25]
```diff
@@ -838,9 +1213,9 @@ definition {
...
 	@priority = "3"
-	test CannotAddToCartMultipleProductQuantityWhenOrderItemQuantityIsHigherThanStockQuantity {
+	test CannotAddToCartMultipleProductQuantityWhenOrderItemQuantityIsHigherThanStockQuantityAndBackOrdersIsUnable {
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase:1218]:
```suggestion
	test CannotAddToCartMultipleQuantityWhenOrderItemQuantityIsHigherThanStockQuantityAndBackOrderIsDisabled {
```

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase [hunk 26]
```diff
@@ -904,9 +1277,9 @@ definition {
 		}
 	}
 
-	@description = "This is a test for COMMERCE-10559. Multiple quantity can be added to cart until there's stock quantity "
+	@description = "This is a test for COMMERCE-10559. Multiple quantity can be added to the cart until there's stock quantity and back orders is unabled"
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase:1280]:
```suggestion
	@description = "This is a test for COMMERCE-10559. Multiple quantities can be added to the cart until there's stock quantity and back order is disabled"
```

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase [hunk 27]
```diff
@@ -904,9 +1277,9 @@ definition {
...
 	@priority = "3"
-	test CannotAddToCartMultipleProductQuantityWhenStockQuantityIsLess {
+	test CannotAddToCartMultipleProductQuantityWhenStockQuantityIsLessAndBackOrdersIsUnable {
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase:1282]:
```suggestion
	test CannotAddToCartMultipleQuantityWhenStockQuantityIsLessAndBackOrderIsDisabled {
```

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase [hunk 28]
```diff
@@ -993,7 +1364,7 @@ definition {
 		}
 	}
 
-	@description = "This is a test for COMMERCE-10491. The product can be added to cart until there's stock quantity"
+	@description = "This is a test for COMMERCE-10491. The product can be added to the cart until there's stock quantity and back orders is unabled"
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase:1367]:
```suggestion
	@description = "This is a test for COMMERCE-10491. The product can be added to the cart until there's stock quantity and back order is disabled"
```

**Review by lorenzolapescara — CHANGES_REQUESTED** [review]

**Comment by Brunno0liveira** [comment]
ci:test:sf

**Comment by Brunno0liveira** [comment]
ci:test:sf

**Comment by Brunno0liveira** [comment]
ci:test:sf

**Review by lorenzolapescara — APPROVED** [review]

**Pull request closed without merging**